### DISCUSSION TITLE
AMQPUri in deadpool lapin

### DISCUSSION THREAD

**Discussion by metalalive** [bikeshedder/deadpool#332] [Ideas]
Hi, is it possible to pass [`lapin::uri::AMQPUri`](https://docs.rs/amq-protocol-uri/latest/amq_protocol_uri/struct.AMQPUri.html) to [`deadpool_lapin::Config`](https://docs.rs/deadpool-lapin/latest/deadpool_lapin/struct.Config.html) for pool setup ?
currently the only way I found is to specify optional [formatted string](https://www.rabbitmq.com/docs/uri-spec) to `Config::url` ,
 then `lapin` internally extracts the string to `AMQPUri` .
I am not sure if I can configure it directly with `AMQPUri` ?

Thanks

**Comment by bikeshedder** [comment]
Tracking issue: https://github.com/bikeshedder/deadpool/issues/337
